irods/test/data_obj_test.py

```diff
@@ -1463,6 +1465,49 @@ def test_register_with_checksum(self):
         # delete file
         os.remove(test_file)
 
+
+    def test_object_names_with_nonprintable_chars (self):
+        test_dir = helpers.irods_shared_tmp_dir()
+        loc_server = self.sess.host in ('localhost', socket.gethostname())
+        if not(test_dir) and not(loc_server):
+            self.skipTest('data_obj register requires server has access to local or shared files')
+        temp_names = []
+        vault = ''
+        try:
+
+            #import pdb;pdb.set_trace() # dwm_
+
+            fname = re.sub( r'[/\n]', '',
+                            ''.join(map(chr,range(1,128))) )
+            prefix_ = 'object_'  + fname
+            resc_name = 'regWithNonPrintableNamesResc'
+            vault = make_ufs_resc_in_tmpdir(self.sess, resc_name, allow_local = loc_server)
+            def enter_file_into_irods ( session, filename, method, **options ):
+                psr = ET( XML_Parser_Type.QUASI_XML, session.server_version)
+                bound_method = getattr(session.data_objects,method)
+                basename = os.path.basename(filename)
+                logical_path = '/{0.zone}/home/{0.username}/{basename}'.format(session,**locals())
+                bound_method( os.path.abspath(filename), logical_path , **options )
+                d = session.data_objects.get(logical_path) #import pprint;pprint.pprint(vars(d)) # dwm_
+                print ('Truth = ',d.path == logical_path)
```

Comment:
formatting in regards to comma-separated arguments is a little inconsistent.

for example, the commas in lines 1490 and 1492

Reply:
corrected